test/mongodb.test.js

```diff
@@ -1438,6 +1456,127 @@ describe('mongodb connector', function() {
       });
     });
 
+    it('find should be able to query by location via near with maxDistance', function(done) {
+      var coords = { lat: 30.274085, lng: 120.15507000000002 };
+
+      geoDb.autoupdate(function(err) {
+        async.parallel([
+          createPost(false),
+          createPost(false),
+          createPost(false),
+          createPost(true),
+        ], function(err) {
+          should.not.exist(err);
+
+          PostWithLocation.find({
+            where: {
+              location: {
+                near: new GeoPoint(coords),
+                maxDistance: 17000,
+                unit: 'meters',
+              },
+            },
+          }, function(err, results) {
+            should.not.exist(err);
+            should.exist(results);
+            results.length.should.be.equal(3);
+            var dist = 0;
+            results.forEach(function(result) {
+              var currentDist = testUtils.getDistanceBetweenPoints(coords, result.location);
+              currentDist.should.be.aboveOrEqual(dist);
+              currentDist.should.be.belowOrEqual(17);
+              dist = currentDist;
+            });
+            done();
+          });
+        });
+      });
+    });
+
+    it('find should be able to query by location via near with minDistance set', function(done) {
+      var coords = { lat: 30.274085, lng: 120.15507000000002 };
+      geoDb.autoupdate(function(err) {
+        async.parallel([
+          createPost(false),
+          createPost(false),
+          createPost(false),
+          createPost(true),
+        ], function(err) {
+          should.not.exist(err);
+
+          PostWithLocation.find({
+            where: {
+              location: {
+                near: new GeoPoint(coords),
+                minDistance: 17000,
+                unit: 'meters',
+              },
+            },
+          }, function(err, results) {
+            should.not.exist(err);
+            should.exist(results);
+            var dist = 0;
+            results.forEach(function(result) {
+              var currentDist = testUtils.getDistanceBetweenPoints(coords, result.location);
+              currentDist.should.be.aboveOrEqual(dist);
+              dist = currentDist;
+            });
+
+            done();
+          });
+        });
+      });
+    });
+
+    it('find should be able to set unit when  query location via near', function(done) {
+      var coords = { lat: 30.274085, lng: 120.15507000000002 };
+
+      geoDb.autoupdate(function(err) {
+        var queryLocatoin = function(distance, unit, distanceInMeter, numOfResult) {
```

Comment:
Can also declare this helper function outside and share (if necessary). Otherwise, do it inline but as:

```
function queryLocation(...)
```

ie. without the var